nitrite.go

```diff
@@ -196,10 +218,14 @@ func createAWSNitroRoot() *x509.CertPool {
 // is not OK or certificate can't be verified, both Result and error will be
 // set! You can use the SignatureOK field from the result to distinguish
 // errors.
-func Verify(data []byte, options VerifyOptions) (*Result, error) {
-	cose := CosePayload{}
-
-	err := cbor.Unmarshal(data, &cose)
+func Verify(
+	attestationBytes []byte,
+	rootCert RootCertFunc,
```

Comment:
WRT the interface update, I really like the usage of the `VerificationTimeFunc`.  It makes for an easy way to work with content from the attestation bytes.  I am a not sure, however, about the RootCertFunc.  Do you see this as needing to be dynamic based on the contents of the bytes?  From the current functions, it looks like it just adds additional complexity.

Reply:
I'll add a cert provider as discussed

Reply:
done